core/server/suspend.go

```diff
@@ -60,6 +60,8 @@ func (cs *coreServer) ToggleSuspendResource(ctx context.Context, msg *pb.ToggleS
 			return nil, err
 		}
 
+		changeSuspendAnnotations(obj, msg.Suspend, msg.Comment, principal)
+
 		if msg.Suspend {
```

Comment:
A couple of things here...

https://github.com/weaveworks/weave-gitops/blob/35231b877e1af6f53fc5c04b9bb516f87ef8e759/core/server/suspend.go#L15-L19

We could reuse the principal from the first call in the second?

Also, we could include the "principalID" in the `log.Info` so that the user who does it is logged out?

We could also include the name and namespace, and maybe even Kind of the resource, so that the logs would be searchable.

"Who unlocked the GitRepository test/test"?

Bonus points for including the Cluster name :-)

Reply:
the name,namespace, and kind are already included . I'll add the principal to the logs :+1: 
https://github.com/weaveworks/weave-gitops/blob/35231b877e1af6f53fc5c04b9bb516f87ef8e759/core/server/suspend.go#L44-L49